src/Prototype/Database.hs

```diff
@@ -91,23 +96,31 @@ getProfileAndList h namespace listname = do
 
 
 --------------------------------------------------------------------------------
-newTodoLists = newTVar Examples.todoLists
+newTodoLists :: STM (TVar (Map String [TVar TodoList]))
+newTodoLists = do
+  let f (k, v) = mapM newTVar v >>= return . (k,)
+  todoLists <- mapM f Examples.todoLists
+  newTVar (Map.fromList todoLists)
 
+getAllTodoLists :: Handle -> STM [TodoList]
 getAllTodoLists h = do
   lists <- readTVar (hTodoLists h)
-  return (map snd lists)
+  let lists' = Map.toList lists
+  mapM readTVar $ concat $ map snd lists'
 
+getTodoLists :: Handle -> String -> STM [TodoList]
 getTodoLists h namespace = do
   lists <- readTVar (hTodoLists h)
-  return ((map snd . filter f) lists)
-  where f = (namespace ==) . fst
+  case Map.lookup namespace lists of
+    Nothing -> return []
+    Just lists' -> mapM readTVar lists'
```

Comment:
I don't know what your opinion is on this, however; wouldn't this be more succinct? 
```suggestion
  maybe (pure mempty) (mapM readTVar) $ Map.lookup namespace lists 
```

Reply:
I'll follow your advice, I don't have strong feelings about it, but I like the simple repeatability of `case`. Here you suggest `maybe`, above it was `pure $ case`.